src/Compilers/Core/Portable/Operations/OperationNodes.cs

```diff
@@ -8077,6 +8196,230 @@ public override IOperation Guard
+        {
+            get
+            {
+                if (_lazyValue == s_unset)
+                {
+                    var value = CreateValue();
+                    SetParentOperation(value, this);
+                    Interlocked.CompareExchange(ref _lazyValue, value, s_unset);
+                }
+
+                return _lazyValue;
+            }
+        }
+        public override ImmutableArray<ISwitchExpressionArmOperation> Arms
+        {
+            get
+            {
+                if (_lazyArms.IsDefault)
+                {
+                    var arms = CreateArms();
+                    SetParentOperation(arms, this);
+                    ImmutableInterlocked.InterlockedInitialize(ref _lazyArms, arms);
+                }
+
+                return _lazyArms;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Represents a C# switch expression arm.
+    /// </summary>
+    internal abstract partial class BaseSwitchExpressionArmOperation : Operation, ISwitchExpressionArmOperation
+    {
+        public BaseSwitchExpressionArmOperation(
+            ImmutableArray<ILocalSymbol> locals,
+            SemanticModel semanticModel,
+            SyntaxNode syntax,
+            bool isImplicit)
+            : base(OperationKind.SwitchExpressionArm, semanticModel, syntax, type: default, constantValue: default, isImplicit)
+        {
+            this.Locals = locals;
+        }
+        public ImmutableArray<ILocalSymbol> Locals { get; }
+        public abstract IPatternOperation Pattern { get; }
+        public abstract IOperation Guard { get; }
+        public abstract IOperation Value { get; }
+
+        public override IEnumerable<IOperation> Children
+        {
+            get
+            {
+                yield return Pattern;
+                if (Guard != null)
+                    yield return Guard;
+                yield return Value;
+            }
+        }
+        public override void Accept(OperationVisitor visitor)
+        {
+            visitor.VisitSwitchExpressionArm(this);
+        }
+        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
+        {
+            return visitor.VisitSwitchExpressionArm(this, argument);
+        }
+    }
+
+    internal sealed partial class SwitchExpressionArmOperation : BaseSwitchExpressionArmOperation
+    {
+        public SwitchExpressionArmOperation(
+            ImmutableArray<ILocalSymbol> locals,
+            IPatternOperation pattern,
+            IOperation guard,
+            IOperation value,
+            SemanticModel semanticModel,
+            SyntaxNode syntax,
+            bool isImplicit)
+            : base(locals, semanticModel, syntax, isImplicit)
+        {
+            this.Pattern = pattern;
+            SetParentOperation(pattern, this);
+            this.Guard = guard;
+            if (guard != null)
+                SetParentOperation(guard, this);
```

Comment:
>SetParentOperation(guard, this); [](start = 16, length = 32)

Same comment about the pattern of setting the parent before initializing the property. Also, it looks like SetParentOperation is safe to use with null input, the conditional logic is unnecessary and ```this.Guard = SetParentOperation(guard, this);``` would just work.